documentation/debugging.rst

```diff
@@ -0,0 +1,106 @@
+    cmake_bindings/std/complex.cpp:#include <bits/stl_uninitialized.h> // std::__uninitialized_default_1
+    cmake_bindings/std/complex.cpp:#include <bits/stl_uninitialized.h> // std::__uninitialized_default_n
+    cmake_bindings/std/complex.cpp:#include <bits/stl_uninitialized.h> // std::__uninitialized_default_n_1
+    cmake_bindings/std/complex.cpp:#include <bits/stl_uninitialized.h> // std::__uninitialized_default_n_1<false>::__uninit_default_n
+    cmake_bindings/std/complex.cpp:#include <bits/stl_uninitialized.h> // std::__uninitialized_default_n_1<true>::__uninit_default_n
+    cmake_bindings/std/complex.cpp:#include <bits/stl_uninitialized.h> // std::__uninitialized_default_n_a
+    cmake_bindings/std/complex.cpp:#include <bits/stl_uninitialized.h> // std::__uninitialized_default_novalue_1
+    cmake_bindings/std/complex.cpp:#include <bits/stl_uninitialized.h> // std::__uninitialized_default_novalue_n_1
+    cmake_bindings/std/complex.cpp:#include <bits/stl_uninitialized.h> // std::__uninitialized_fill
+    cmake_bindings/std/complex.cpp:#include <bits/stl_uninitialized.h> // std::__uninitialized_fill_n
+    cmake_bindings/std/complex.cpp:#include <bits/stl_uninitialized.h> // std::__uninitialized_move_if_noexcept_a
+    cmake_bindings/std/complex.cpp:#include <bits/stl_uninitialized.h> // std::uninitialized_copy
+  
+The important information in the output is the ``std::`` types/functions without the leading underscores.
+ Those are STL-implementation independent types/functions that should be defined elsewhere, not in the headers from the ``bits`` directory.
+In this particular example, the function of interest is ``std::uninitialized_copy``.  
+
+A quick search in the C++ documentation or in the WWW,  tells that this function is defined in the <memory> header.
+Therefore, this information should be hardcoded into the binder.
+
+
+3.  The internal binder function that handles the STL library mappings is located in ``source/types.cpp``:``add_relevant_include_for_decl``.  
+Briefly, the function has a map with the STL headers and the types those contain. That should look similar to this:
+
+.. code-block:: python
+
+    { "<algorithm>", {"std::move_backward", "std::iter_swap", "std::min"} },
+    { "<exception>", {"std::nested_exception"} }
+
+If there is a need to make a simple change, like in our case,  the map for the ``<memory>`` can be added like this:
+
+
+.. code-block:: python
+
+    { "<algorithm>", {"std::move_backward", "std::iter_swap", "std::min"} },
+    { "<exception>", {"std::nested_exception"} },
+    { "<memory>", {"std::uninitialized_copy"} },
+
+
+4.  After the changes are done, the binder executable should be recompilled and re-used to create the desired bindings. 
+    In some cases, many iterations of the described procedure will be needed till all the STL types/functions will be mapped to the correct includes. 
+    
+    If this fixes your problem please let us know, or make a pull request!
```

Comment:
optional: would it be possible to also add small paragraph on include-mapping? And point reader to https://github.com/RosettaCommons/binder/blob/master/source/type.cpp#L168 map

Reply:
Hi @lyskov ,

1) Done.
2) please note that this request includes the original commits from 
@danpf , so one can  merge the #58 and then this request. 

Best regards,
Andrii

Reply:
Will do, thank you @andriish !